x-pack/plugins/apm/public/components/routing/templates/apm_service_template/index.tsx

```diff
@@ -123,12 +124,43 @@ function TemplateWithContext({
   );
 }
 
+export function isMetricsTabHidden({
+  agentName,
+  serviceRuntimeName,
+}: {
+  agentName?: string;
+  serviceRuntimeName?: string;
+}) {
+  return (
+    !agentName ||
+    isRumAgentName(agentName) ||
+    isJavaAgentName(agentName) ||
+    isIosAgentName(agentName) ||
+    isJRubyServiceRuntimeName(serviceRuntimeName)
+  );
+}
```

Comment:
Personal preference: I find it easier to reason with helper `(...) => boolean` functions when they are named in the positive: i.e. `isMetricsTabHidden` -> `isMetricsTabShown`, `hidden: !isMetricsTabShown(...)`. I don't feel too strongly, so apply at your discretion.